File: n8n-setup-scripts/import-to-db.py
#!/usr/bin/env python3

import json
import sqlite3
import os
from datetime import datetime
from pathlib import Path

# N8n database location
N8N_HOME = os.path.expanduser('~/.n8n')
N8N_DB = os.path.join(N8N_HOME, 'database.sqlite')

def load_workflow(filepath):
    """Load workflow JSON file"""
    with open(filepath, 'r') as f:
        return json.load(f)

def import_workflow_to_db(db_path, workflow_data):
    """Import workflow directly into n8n database"""

    if not os.path.exists(db_path):
        print(f"❌ Database not found: {db_path}")
        print(f"   Make sure n8n has been started at least once")
        return False

    try:
        conn = sqlite3.connect(db_path)
        cursor = conn.cursor()

        now = datetime.now().isoformat()

        # Insert workflow
        workflow_name = workflow_data.get('name', 'Untitled Workflow')
        nodes = json.dumps(workflow_data.get('nodes', []))
        connections = json.dumps(workflow_data.get('connections', {}))
        active = workflow_data.get('active', False)

        cursor.execute('''
            INSERT INTO workflow (name, nodes, connections, active, createdAt, updatedAt)
            VALUES (?, ?, ?, ?, ?, ?)
        ''', (workflow_name, nodes, connections, 1 if active else 0, now, now))

        conn.commit()
        workflow_id = cursor.lastrowid

        print(f"✅ Imported: {workflow_name} (ID: {workflow_id})")

        conn.close()
        return True

    except Exception as e:
        print(f"❌ Error importing workflow: {e}")
        return False

def main():
    print("═════════════════════════════════════════════════")
    print("    N8n Workflow Database Import")
    print("═════════════════════════════════════════════════\n")

    # Get workflow files
    workflows_dir = Path(__file__).parent.parent / 'n8n-workflows'
    workflow_files = sorted(workflows_dir.glob('*-workflow.json'))

    if not workflow_files:
        print("❌ No workflow files found in n8n-workflows/")
        return 1

    print(f"Found {len(workflow_files)} workflow(s)\n")

    # Check database
    if not os.path.exists(N8N_DB):
        print(f"❌ N8n database not found: {N8N_DB}")
        print("   Start n8n first: n8n start")
        return 1

    print(f"Using database: {N8N_DB}\n")

    # Import each workflow
    imported_count = 0
    for workflow_file in workflow_files:
        try:
            workflow_data = load_workflow(workflow_file)
            if import_workflow_to_db(N8N_DB, workflow_data):
                imported_count += 1
        except Exception as e:
            print(f"❌ Failed to process {workflow_file.name}: {e}")

    print(f"\n═════════════════════════════════════════════════")
    print(f"✅ Import complete: {imported_count} workflow(s) imported")
    print("═════════════════════════════════════════════════\n")

    print("Next steps:")
    print("1. Restart n8n: Ctrl+C to stop, then run 'n8n start'")
    print("2. Open http://localhost:5678")
    print("3. Check the Workflows section to see your imported workflows")
    print("4. Activate them and start using!\n")

    return 0

if __name__ == '__main__':
    exit(main())

File: n8n-setup-scripts/import-to-db.js
#!/usr/bin/env node

/**
 * N8n Workflow Database Import
 * Directly inserts workflows into n8n's SQLite database
 */

const fs = require('fs');
const path = require('path');
const sqlite3 = require('sqlite3').verbose();
const os = require('os');

const N8N_HOME = path.join(os.homedir(), '.n8n');
const N8N_DB = path.join(N8N_HOME, 'database.sqlite');

/**
 * Load workflow JSON
 */
function loadWorkflow(filepath) {
  try {
    const content = fs.readFileSync(filepath, 'utf8');
    return JSON.parse(content);
  } catch (error) {
    console.error(`Failed to load workflow: ${error.message}`);
    return null;
  }
}

/**
 * Import workflow to database
 */
async function importWorkflowToDb(dbPath, workflowData) {
  return new Promise((resolve, reject) => {
    if (!fs.existsSync(dbPath)) {
      console.log(`❌ Database not found: ${dbPath}`);
      console.log('   Make sure n8n has been started at least once');
      resolve(false);
      return;
    }

    const db = new sqlite3.Database(dbPath, (err) => {
      if (err) {
        console.error(`Failed to connect to database: ${err.message}`);
        resolve(false);
        return;
      }

      const now = new Date().toISOString();
      const name = workflowData.name || 'Untitled Workflow';
      const nodes = JSON.stringify(workflowData.nodes || []);
      const connections = JSON.stringify(workflowData.connections || {});
      const active = workflowData.active ? 1 : 0;

      // Insert workflow
      db.run(
        `INSERT INTO workflow (name, nodes, connections, active, createdAt, updatedAt)
         VALUES (?, ?, ?, ?, ?, ?)`,
        [name, nodes, connections, active, now, now],
        function(err) {
          if (err) {
            console.error(`❌ Failed to insert workflow: ${err.message}`);
            db.close();
            resolve(false);
          } else {
            console.log(`✅ Imported: ${name} (ID: ${this.lastID})`);
            db.close();
            resolve(true);
          }
        }
      );
    });
  });
}

/**
 * Main setup
 */
async function main() {
  console.log('═════════════════════════════════════════════════');
  console.log('    N8n Workflow Database Import');
  console.log('═════════════════════════════════════════════════\n');

  // Get workflow files
  const workflowsDir = path.join(__dirname, '..', 'n8n-workflows');
  let workflowFiles = [];

  try {
    workflowFiles = fs.readdirSync(workflowsDir)
      .filter(f => f.endsWith('-workflow.json'))
      .map(f => path.join(workflowsDir, f));
  } catch (error) {
    console.error(`❌ Failed to read workflows directory: ${error.message}`);
    return 1;
  }

  if (workflowFiles.length === 0) {
    console.log('❌ No workflow files found in n8n-workflows/');
    return 1;
  }

  console.log(`Found ${workflowFiles.length} workflow file(s)\n`);

  // Check database exists
  if (!fs.existsSync(N8N_DB)) {
    console.log(`❌ N8n database not found: ${N8N_DB}`);
    console.log('   Start n8n first: n8n start');
    return 1;
  }

  console.log(`Using database: ${N8N_DB}\n`);

  // Import each workflow
  let importedCount = 0;
  for (const workflowFile of workflowFiles) {
    const workflowData = loadWorkflow(workflowFile);
    if (workflowData) {
      const success = await importWorkflowToDb(N8N_DB, workflowData);
      if (success) importedCount++;
    }
  }

  console.log(`\n═════════════════════════════════════════════════`);
  console.log(`✅ Complete: ${importedCount}/${workflowFiles.length} workflows imported`);
  console.log('═════════════════════════════════════════════════\n');

  console.log('⚠️  Important: You must RESTART n8n for changes to take effect!\n');

  console.log('Next steps:');
  console.log('1. Stop n8n (Ctrl+C)');
  console.log('2. Start n8n again: n8n start');
  console.log('3. Open http://localhost:5678');
  console.log('4. Your workflows will appear in the Workflows section');
  console.log('5. Activate them and start using!\n');

  return 0;
}

// Check if sqlite3 is installed
try {
  require('sqlite3');
} catch (error) {
  console.error('❌ sqlite3 module not found');
  console.error('Install with: npm install sqlite3');
  process.exit(1);
}

main().then(code => process.exit(code));
